Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1

```diff
@@ -0,0 +1,358 @@
+                $UserInput = Read-Host -Prompt "(Y|N)"
+                if($UserInput -ne "Y")
+                {
+                    return;
+                }
+            }
+            
+            Write-Host "`t"
+            Write-Host "Disabling RDP access on [$($totalCloudServiceWithEnableRDPAccess)] cloud service(s)..."
+            try 
+            {
+                $cloudServiceWithEnabledRDPAccess | ForEach-Object {
+                    $serviceName = $_.CloudServiceName
+                    $rgName = $_.ResourceGroupName
+                    
+                    # Disabling RDP access
+                    $_.RDPExtensionDetails | ForEach-Object {
+                        
+                        # TODO: Add 'UninstallConfiguration' switch to remove extension configurations associated with the service 
+                        $output = Remove-AzureServiceRemoteDesktopExtension -ServiceName $serviceName -Slot $_.Slot -Role $_.Role -ErrorAction SilentlyContinue
+                        if($null -eq $output -and $output.OperationStatus -ine "Succeeded")
+                        {
+                            $item =  New-Object psobject -Property @{  
+                                CloudServiceName = $serviceName                
+                                ResourceGroupName = $rgName
+                            }
+    
+                            $skippedCloudServiceFromRemediation += $item
+                            break;
+                        }
+                    }
+
+                    if(($skippedCloudServiceFromRemediation | Measure-Object).Count -eq 0)
+                    {
+                        $_ | Select-Object @{Expression={($_.ResourceGroupName)};Label="ResourceGroupName"},@{Expression={$_.CloudServiceName};Label="CloudServiceName"} | Sort-Object | Format-Table | Out-String
+                    }
+                }
+            }
+            catch
+            {
+                $item =  New-Object psobject -Property @{  
+                    CloudServiceName = $_.CloudServiceName                
+                    ResourceGroupName = $_.ResourceGroupName
+                }
+                $skippedCloudServiceFromRemediation += $item
+            }
+        }
+        else
+        {
+            Write-Host "No cloud service(s) found with enabled RDP access." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::DoubleDashLine)
+            break
+        }  
+    }
+    catch {
+        Write-Host "Error occured while performing remediation. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+        break
+    }
+
+    if(($skippedCloudServiceFromRemediation | Measure-Object).Count -eq 0)
+    {
+        Write-Host "Successfully disabled RDP access from cloud service(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else 
+    {
```

Comment:
In this case for some of the cloud services we might have be able to remove RDP extensions we should print those as well.